packages/secure_storage/amplify_secure_storage_dart/lib/src/types/macos_secure_storage_options.dart

```diff
@@ -0,0 +1,57 @@
+// Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/// {@template amplify_secure_storage_dart.macos_secure_storage_options}
+/// Configuration options that are specific to MacOS.
+/// {@endtemplate}
+class MacOSSecureStorageOptions {
+  /// {@macro amplify_secure_storage_dart.macos_secure_storage_options}
+  const MacOSSecureStorageOptions({
+    required this.useDataProtection,
+    this.accessGroup,
+  }) : assert(
+          useDataProtection || accessGroup == null,
+          'useDataProtectionKeychain must be set to true when using an access group.',
+        );
+
+  /// The default options for MacOS.
+  static const defaultOptions = MacOSSecureStorageOptions(
+    useDataProtection: true,
+  );
+
+  /// Sets the `kSecUseDataProtectionKeychain` attribute to true for
+  /// all Keychain operations.
+  ///
+  /// Reference: [kSecUseDataProtectionKeychain](https://developer.apple.com/documentation/security/ksecusedataprotectionkeychain?language=objc)
+  ///
+  /// Set this to true to enforce that Keychain items behave
+  /// like iOS Keychain items.
+  ///
+  /// **Note:** This must be set to true when specifying an [accessGroup].
+  ///
+  /// **Note:** Setting this to true requires that the app is added to one
+  /// or more Keychain Access Groups. See the section titled
+  /// "Add Apps to One or More Keychain Access Groups" at the
+  /// link below for more info.
+  ///
+  /// Reference: [Sharing Access to Keychain Items Among a Collection of Apps](https://developer.apple.com/documentation/security/keychain_services/keychain_items/sharing_access_to_keychain_items_among_a_collection_of_apps?language=objc)
+  final bool useDataProtection;
```

Comment:
Okay, I think we just need to update the notes here, then. You say "Setting this to true requires that the app is added to one or more Keychain Access Groups" but I think that's a bit misleading, since the app always defaults to the package ID as its access group, right? If you specify `useDataProtection = true, accessGroup = null` this should still work as I understand because accessGroup defaults to packageId.

Also, should this line say "should" instead of "must"? 

> `**Note:** This must be set to true when specifying an [accessGroup].`

And can we change the constructor to:

```dart
const MacOSSecureStorageOptions({
    this.useDataProtection = true,
    this.accessGroup,
  });
```

Overall, I'm still really not seeing a good reason to have this be false outside of tests - the Keychain docs even say:

> It’s highly recommended that you set the value of this key to true for all keychain operations. This key helps to improve the portability of your code across platforms. Use it unless you specifically need access to items previously stored in a legacy keychain in macOS.

If we _just_ wanted it for tests, we could use Zone overrides instead.

Reply:
> the app always defaults to the package ID as its access group, right?

Not really on MacOS. You need to enable the Keychain Sharing capability in Xcode for MacOS apps. It is not enabled by default. Once you enable it, it will automatically be added to one group, which defaults to the Team ID + Package ID.

> If you specify useDataProtection = true, accessGroup = null this should still work as I understand because accessGroup defaults to packageId

If you have enabled the Keychain Sharing capability in Xcode, yes. 

I could update this to note that you have to enable the Keychain Sharing capability. That is pretty much what the docs at located at the link says to do. I would prefer to just point to Apple documentation in case this changes in a future version of Xcode or something. Another option would be to point to docs.amplify.aws. I think that could be a reasonable update later on. I have a TODO to update the message in the exception that is thrown when someone has this set to true without enabling Keychain Sharing to include a link to our docs. I could potentially link to that from here once it exists.

> Overall, I'm still really not seeing a good reason to have this be false outside of tests

Aside from our owns tests, here are three reasons:
1. Less friction for someone that is just testing out Amplify & Flutter. I believe you will need an Apple developer account to set this to true, since enabling Keychain Sharing requires setting up code signing.
2. Testing for customers. Just as it is a pain to set up code signing within our integration tests, it could be for customers.
3. You might need it to be false if you were making a CLI tool or something else non-flutter. Not really a use case we are trying to support though.

Reply:
Ah okay yeah I didn’t really you had to enable it. Makes sense to me. Thanks for the explanation 👍